experimental/oauther/oauth2_complete.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package oauther
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+func (o *oAuther) oauth2Complete(w http.ResponseWriter, r *http.Request) {
+	authedUserID := r.Header.Get("Mattermost-User-ID")
```

Comment:
As an aside, I'd love to be able to reuse `http.MattermostAuthorizationRequired` as exposed by this package. (Our `http` package in this repository might also be the long-term home for this code.)

Reply:
That sounds like a great idea, but maybe we should put those in a http package and implement them on a different PR.